apps/wallet/src/shared/validation.ts

```diff
@@ -58,38 +58,73 @@ export function createTokenValidation(
-        )
-        .test(
-            'gas-balance-check',
-            `Insufficient ${GAS_SYMBOL} balance to cover gas fee (${formatBalance(
-                gasBudget,
-                gasDecimals
-            )} ${GAS_SYMBOL})`,
-            (amount?: BigNumber) => {
+        .test({
+            name: 'gas-balance-check-enough-single-coin',
+            test: function (_, ctx) {
+                // ignore gas budget check if gasBudget is null or gasInputBudgetEst is not null
+                if (!ctx.parent?.gasInputBudgetEst && !gasBudget) {
+                    return ctx.createError({
+                        message: `Insufficient ${GAS_SYMBOL}, there is no individual coin with enough balance to cover for the gas Fee`,
+                    });
+                }
+                const gasBudgetInput = (ctx.parent?.gasInputBudgetEst ??
+                    gasBudget) as number;
+
+                if (maxSuiSingleCoinBalance >= gasBudgetInput) {
+                    return true;
+                }
+
+                return ctx.createError({
+                    message: `Insufficient ${GAS_SYMBOL}, there is no individual coin with enough balance to cover for the gas fee (${formatBalance(
+                        gasBudgetInput,
+                        gasDecimals
+                    )} ${GAS_SYMBOL})`,
+                });
+            },
+        })
+
+        .test({
+            name: 'gas-balance-check',
+            test: function (amount: BigNumber | undefined, ctx) {
+                // For Pay All SUI and SUI coinType, we don't need to check gas balance.
+                if (ctx.parent?.isPayAllSui && coinType === GAS_TYPE_ARG) {
+                    return true;
+                }
+
+                // this should not happen since we have a required validation
                 if (!amount) {
                     return false;
                 }
+                // check updated gas balance base on gasInputBudgetEst
                 try {
                     let availableGas = gasBalance;
                     if (coinType === GAS_TYPE_ARG) {
                         availableGas -= BigInt(
                             amount.shiftedBy(decimals).toString()
                         );
                     }
-                    return availableGas >= gasBudget;
+                    if (
+                        availableGas >=
+                        (ctx.parent?.gasInputBudgetEst || gasBudget)
+                    ) {
+                        return true;
+                    }
+                    return ctx.createError({
+                        message: `Insufficient ${GAS_SYMBOL} balance to cover gas fee (${formatBalance(
+                            ctx.parent?.gasInputBudgetEst || gasBudget,
+                            gasDecimals
+                        )} ${GAS_SYMBOL})`,
+                    });
                 } catch (e) {
-                    return false;
+                    return ctx.createError({
```

Comment:
... and move this one outside of catch. If everything went well we return true at line 110 otherwise the error